kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/conversion/ConversionUtil.java

```diff
@@ -4,13 +4,15 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.BiFunction;
 import java.util.function.Function;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import javafx.util.Pair;
 
 public class ConversionUtil {
 
-  private final Map<Class, List<Pair<Pattern, Function<String, ?>>>> typeConversionMap = new HashMap<>();
+  private final Map<Class, List<Pair<Pattern, BiFunction<String, Matcher, ?>>>> typeConversionMap = new HashMap<>();
 
   public static Object convertFieldValue(String fieldPath, String fieldValue,
       Map<String, Function<String, Object>> conversionMap) {
```

Comment:
Why do we store registered conversion functions, and also supply conversion functions on the converter method?
I probably need to do a more thorough reading of the code.

Reply:
Can register conversions based on type (I.e. all LocalDates should be parsed like this...), and more granularly at the field level (I.e. this specific field needs to be parsed like this...)